lib/web/machineid.go

```diff
@@ -262,6 +263,45 @@ type updateBotRequest struct {
 	Roles []string `json:"roles"`
 }
 
+// getBotInstance retrieves a bot instance by id
+func (h *Handler) getBotInstance(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (any, error) {
+	botName := p.ByName("name")
+	instanceId := p.ByName("id")
+	if botName == "" {
+		return nil, trace.BadParameter("empty bot name")
+	}
+	if instanceId == "" {
+		return nil, trace.BadParameter("empty id")
+	}
+
+	clt, err := sctx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	instance, err := clt.BotInstanceServiceClient().GetBotInstance(r.Context(), &machineidv1.GetBotInstanceRequest{
+		InstanceId: instanceId,
+		BotName:    botName,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err, "error querying bot instance")
+	}
+
+	yaml, err := yaml.Marshal(instance)
+	if err != nil {
+		return nil, trace.Wrap(err, "error stringifying to yaml")
+	}
+
+	return GetBotInstanceResponse{
+		BotInstance: instance,
+		Yaml:        string(yaml),
+	}, nil
+}
+
+type GetBotInstanceResponse struct {
+	BotInstance *machineidv1.BotInstance `json:"bot_instance"`
+	Yaml        string                   `json:"yaml"`
```

Comment:
Minor style nit: In Go, we tend to fully capitalize initialisms/acronyms e.g `YAML`